File: LeetCode/87/main.cpp
class Solution {
public:
  int evalRPN(vector<string> &arr) {
    int n = arr.size();
    stack<int> st;

    for (int i = 0; i < n; i++) {
      string ch = arr[i];

      if (ch.length() == 1 && int(ch[0]) < 48) {
        int b = st.top();
        st.pop();
        int a = st.top();
        st.pop();
        if (ch == "+") {
          st.push(a + b);
        } else if (ch == "-") {
          st.push(a - b);
        } else if (ch == "*") {
          st.push(a * b);
        } else if (ch == "/") {
          st.push(a / b);
        }

      } else {
        int num = stoi(ch);
        st.push(num);
      }
    }
    return st.top();
  }
};

File: LeetCode/87/main.py
class Solution:
    def evalRPN(self, arr: List[str]) -> int:

        st = []

        for s in arr:
            if len(s) == 1 and s in '+-*/':
                b = st.pop()
                a = st.pop()

                if (s == '+'):
                    st.append(a + b)
                elif (s == '-'):
                    st.append(a - b)
                elif (s == '*'):
                    st.append(a * b)
                elif (s == '/'):
                    st.append(int(a / b))

            else:
                st.append(int(s))

        return st.pop()
